docfx_project/articles/bootstrapping-trust/part-01.encryption-everywhere.md

```diff
@@ -0,0 +1,310 @@
+
+The solution can be improved. Secrets could be generated per software
+component. The controller, each router, and each SDK client could have a
+unique secret. This secret would then individually identity each
+component! It is a significant improvement, but how does each component
+verify connections? Do they challenge for the incoming connections
+secret and compare it to a list? That means that a pair of systems that
+need to connect must have each other's secrets. Secret sharing will not
+do! We can not be copying secrets between every machine. One machine
+that is compromised would mean that many secrets will are revealed!
+
+This solution can be evolved and improved, but we do not have to it!
+Where this line of thought goes is right into existing technology that
+has already been solved for us! It is call (public-key
+cryptography)[https://en.wikipedia.org/wiki/Public-key_cryptography],
+and it provides everything we need.
+
+Public-key cryptography allows each device to have a unique secret,
+private key, that prooves its unique identity. That private key is
+mathematically tied to a public key. The public key can be used to
+encrypt messages that only the private key holder can decrypt. Also, the
+public key cannot be used to derive the original private key. This
+functionality fits perfectly with what our distributed system needs!
+Alas, public-key cryptography introduces complex behaviors, setup , and
+management. In the next article, we will dive a little deeper into this
+topic. For now, let us take it on faith that it will serve our needs
+well.
+
+#### Setting It Up
+
+So we have decided that public-key cryptography is the answer. What does
+that mean? What do I have to do? Let us explore what would need to be
+done by a human or a piece of software automating this process. Don't
+worry if you don't get all of this; the gist is all you need for now.
+Later parts will expand upon this terminology. In fact, after reading
+the later parts, consider revisiting this part.
+
+Consider the following diagram of a "mesh" distributed system. This mesh
+could be any type of system such as a mesh of Ziti Routers, or maybe it
+is a system of sensors on an airplane. What they do does not matter.
+What matters is that this system has multiple pieces of software
+connecting amongst themselves. Consider what it means to accomplish this
+using public-key cryptography.
+
+![Image of a simple mesh](./images/mesh.png)
+
+In the diagram above, each system needs:
+
+- a key pair for client & server connections
+- to have the public keys of each system it is connecting to
+
+So what do we need to do? Drop into a CLI and start generating keys on
+each machine. Do that by using these commands:
+
+```
+openssl ecparam -name secp256k1-genkey -param_enc explicit -out private-key.pem
+```
+
+```
+openssl req -new -x509 -key private-key.pem -out server.pem -days 360
+```
+
+Voila - you now have a self-signed certificate (that no one trusts). You
```

Comment:
this is the first time our newbie has read the term "self-signed certificate" and what that means. you also describe them in section 4 - maybe just a pointer that they'll learn what a "self-signed certificate" is in part 4?

Reply:
Added and reworded.